### DISCUSSION TITLE
React version support

### DISCUSSION THREAD

**Discussion by MADHUMITHA-GANAPATHY** [grafana/faro-web-sdk#478] [Q&A]
What versions of react and angular does Faro support ? Are there restrictions with using older versions?

**Comment by codecapitano** [comment]
Hi @MADHUMITHA-GANAPATHY 

I am not aware of any restrictions. 

Two things to consider:

**1. When using a (very) old Node version to build**
If you are using the npm version you may run into issues when using older node versions [1].

Using the CDN version should work without any problems.



**2. When using the React Instrumentation package**
Faro provides a React instrumentation package. This has issues with the most recent version of the React router [2].

[1: Grafana Faro Version Support Question](https://github.com/grafana/faro-web-sdk/discussions/395)
[2: FaroRoutes support for RouterProvider react router v6](https://github.com/grafana/faro-web-sdk/issues/235)

Cheers,
Marco

**Reply by fnordian** [reply]
That is a very vague response. Is there a page that outlines what functionality is available with which version? I am trying to integrate faro sdk in a react web app and it's very unreliable. It is very unclear to me how to me if and how auto instrumentation should work. There are bits and pieces of documentation and references to opentelemetry. But something is missing, and things only work sort of, e.g. traces are missing or not properly separated.

Faro promises all those tracing features, but after spending days on setting up infrastructure and integrations, it leaves one on its own in a jungle of open or abandoned bugs, SO hacks and references to 3rd party projects.

**Comment by codecapitano** [comment]
Hi @fnordian 

> That is a very vague response. Is there a page that outlines what functionality is available with which version? 

You mean which Faro version contains a certain set of features or version support for different libraries?
For the first we only have the changelog, for the latter we don't have any support matrix atm.

> It is very unclear to me how to me if and how auto instrumentation should work.

Did you already had a look at the [docs for instrumentations](https://grafana.com/docs/grafana-cloud/monitor-applications/frontend-observability/faro-web-sdk/components/provided-instrumentations/), maybe this can answer some questions already?

We are working on restructuring and updating our docs so content is either to find.

> There are bits and pieces of documentation and references to opentelemetry. But something is missing, and things only work sort of, e.g. traces are missing or not properly separated.

So you use Faro together with the web-tracing package?

> Faro promises all those tracing features,
Which features do you mean which are advertised but not working so we can have a look at it.


cc @cedricziel
